### PR TITLE
Payment requests clientside

### PR DESCRIPTION
**Pull request by zapu**
This PR introduces:
- API support for `stellar/submitrequest` and `stellar/requestdetails` endpoints.
- RPC that prepare Stellar request, and call the API; as well as RPC that fetches request and "localizes" it for use in rendering.
- Chat message type for requests and cli rendering for that message.
- Small test for creating request that mocks out the APIs and returns random `KeybaseRequestID`. The test checks if chat message has been sent.

### CODE REVIEW

File: go/stellar/stellar.go [hunk 1]
```diff
@@ -1053,10 +1058,85 @@ func ChatSendPaymentMessage(m libkb.MetaContext, recipient stellarcommon.Recipie
...
+		}
+	}
+
+	// Make sure chat is functional. Chat message is the only way for
+	// the recipient to learn about the request, so it's essential
+	// that we are able to send REQUESTPAYMENT chat message.
+	m.G().StartStandaloneChat()
 	if m.G().ChatHelper == nil {
-		return errors.New("cannot send SendPayment message:  chat helper is nil")
+		return ret, errors.New("cannot send SendPayment message: chat helper is nil")
```
Comment (patrickxb) [inline go/stellar/stellar.go]: cannot send RequestPayment message

File: go/chat/msgchecker/boxed_checker.go [hunk 2]
```diff
@@ -59,6 +59,9 @@ func checkMessageBoxedLength(msg chat1.MessageBoxed) error {
 	case chat1.MessageType_SENDPAYMENT:
 		return boxedFieldLengthChecker("SENDPAYMENT message", len(msg.BodyCiphertext.E),
 			BoxedSendPaymentMessageBodyMaxLength)
+	case chat1.MessageType_REQUESTPAYMENT:
```
Comment (patrickxb) [inline go/chat/msgchecker/boxed_checker.go]: you're going to have to vendor chat/msgchecker into keybase/keybase/go so gregord can handle these messages

File: protocol/avdl/stellar1/local.avdl [hunk 3]
```diff
@@ -353,4 +383,7 @@ protocol local {
 
   map<OutsideCurrencyCode, OutsideCurrencyDefinition> getAvailableLocalCurrencies();
   string formatLocalCurrencyString(string amount, OutsideCurrencyCode code);
+
+  void sendRequestCLILocal(string recipient, union { null, Asset } asset,
```
Comment (patrickxb) [inline protocol/avdl/stellar1/local.avdl]: since `send` could be confused with sending money, what about changing this to `makeRequestCLILocal()`?

**Comment by patrickxb** [comment]
Looks good...can you show what CLI rendering of these messages looks like?  Thanks!

File: go/chat/msgchecker/constants.go [hunk 4]
```diff
@@ -8,20 +8,22 @@ const (
 	ReactionMessageMaxLength = 50
 	HeadlineMaxLength        = 280
 	TopicMaxLength           = 20
+	PaymentTextMaxLength     = 240
```
Comment (zapu) [inline go/chat/msgchecker/constants.go]: Do we want to keep this? It's for checking ``Requestpayment.Note` length. We can use `TextMessageMaxLength` for that.

**Comment by zapu** [comment]
thanks! CLI rendering:
```
[1]  [tuser719edc 22s]   requested Lumens worth 10 USD (54.7336387 XLM)                                  
[2]  [tuser719edc 13s]   requested 20 XLM                                                                
                         > hello                                                                         
```

File: go/chat/msgchecker/constants.go (hunk 4 above)
Comment (patrickxb) [reply go/chat/msgchecker/constants.go]: Seems fine to keep it.

**Comment by patrickxb** [comment]
Nice! Ok I'm ready to approve.  Let's coordinate with @mmaxim to vendor the msgchecker into gregor and deploy it.

File: protocol/avdl/chat1/local.avdl [hunk 5]
```diff
@@ -175,6 +175,11 @@ protocol local {
     string kbTxID;
   }
 
+  record MessageRequestPayment {
+    string requestID;
+    string note;
```
[the remark this answers is not in the document]
Comment (patrickxb) [reply protocol/avdl/chat1/local.avdl]: I like `note`

**Pull request merged**